tools/clang/test/HLSLFileCheck/hlsl/semantics/legacy/uniform_in_entry_point_ps.hlsl

```diff
@@ -0,0 +1,43 @@
+// RUN: %dxc -D PARAMTYPE=Tex2D /E main_fragment -T ps_6_0 %s | FileCheck %s -check-prefix=CHECK1
+// CHECK1: warning: effect state block ignored - effect syntax is deprecated. To use braces as an initializer use them with equal signs. [-Weffects-syntax]
+// CHECK1: attribute uniform only valid for non-entry-point functions.
+
+// RUN: %dxc -D PARAMTYPE=RW /E main_fragment -T ps_6_0 %s | FileCheck %s -check-prefix=CHECK2
+// CHECK2: warning: effect state block ignored - effect syntax is deprecated. To use braces as an initializer use them with equal signs. [-Weffects-syntax]
+// CHECK2: attribute uniform only valid for non-entry-point functions.
+
+
+// RUN: %dxc -D PARAMTYPE=SamplerState /E main_fragment -T ps_6_0 %s | FileCheck %s -check-prefix=CHECK3
```

Comment:
This pattern for varying type could be adopted for prior test as well, though I'm not sure how much value the extra level of indirection through "Tex2D" provides.  Also, if the expected errors are the same, there's no need to have independent CHECK prefixes and separate blocks to check each case.